### ISSUE TITLE
feat: add api and documentation using swagger

### ISSUE THREAD

**Issue by manuel-rw** [homarr-labs/homarr#483]
- Add API endpoints to all procedures that should be accessible via REST
- Add a swagger documentation
- Add a new page for the swagger docs

Out of scope:
- API auth tokens
- Subscriptions

**Comment by manuel-rw** [comment]
Blocked by https://github.com/jlalmes/trpc-openapi/issues/442 / https://github.com/Vercjames/package-trpc-swagger/issues/11

**Comment by manuel-rw** [comment]
Also blocked by https://github.com/swagger-api/swagger-ui/issues/10045
